File: miniapp/h5/pack.py
import os
import zipfile

ROOT = os.path.dirname(os.path.abspath(__file__))
DIST = os.path.join(ROOT, '..', 'dist-h5')
OUT = os.path.join(ROOT, '..', 'gomoku-minitool.zip')

with zipfile.ZipFile(OUT, 'w', zipfile.ZIP_DEFLATED) as zf:
    for base, _, files in os.walk(DIST):
        for name in sorted(files):
            path = os.path.join(base, name)
            zf.write(path, os.path.relpath(path, DIST))
print(OUT)

File: miniapp/targets/h5/scripts/pack.ts
import { readdirSync, readFileSync, writeFileSync } from 'node:fs'
import { join, relative, resolve } from 'node:path'
import { zipSync } from 'fflate'

const dist = resolve(import.meta.dirname, '../dist')
const out = resolve(import.meta.dirname, '../gomoku-minitool.zip')

const files: Record<string, Uint8Array> = {}
for (const entry of readdirSync(dist, { recursive: true, withFileTypes: true })) {
  if (!entry.isFile()) continue
  const path = join(entry.parentPath, entry.name)
  files[relative(dist, path).split('\\').join('/')] = readFileSync(path)
}

writeFileSync(out, zipSync(files, { level: 9 }))
console.log(out)
